src/shogun/base/MockObject.h

```diff
@@ -0,0 +1,31 @@
+#include <shogun/base/SGObject.h>
```

Comment:
the mock object does not live in Shoguns main src.
It lives in the same directory as your unit test